src/lib/interactionManager/credentialOfferFlow.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  JWTEncodable,
+} from 'jolocom-lib/js/interactionTokens/JSONWebToken'
+import { CredentialOfferRequest } from 'jolocom-lib/js/interactionTokens/credentialOfferRequest'
+import { InteractionType } from 'jolocom-lib/js/interactionTokens/types'
+import { CredentialsReceive } from 'jolocom-lib/js/interactionTokens/credentialsReceive'
+import { SignedCredentialWithMetadata } from './types'
+import { Interaction } from './interaction'
+import { CredentialOfferResponse } from 'jolocom-lib/js/interactionTokens/credentialOfferResponse'
+import { Flow } from './flow'
+import { last } from 'ramda'
+
+type ValidationErrorMap = {
+  invalidIssuer?: boolean
+  invalidSubject?: boolean
+}
+
+export type OfferWithValidity = SignedCredentialWithMetadata & {
+  validationErrors: ValidationErrorMap
+}
+
+export class CredentialOfferFlow extends Flow {
+  public credentialOfferingState: OfferWithValidity[] = []
+  public constructor(ctx: Interaction) {
+    super(ctx)
+  }
+
+  public getState() {
+    return this.credentialOfferingState
+  }
+
+  // TODO Go back to JSONWebToken<JWTEncodable> and use guard functions when casting
+  public async handleInteractionToken(
+    token: JWTEncodable,
+    messageType: InteractionType
+  ): Promise<any> {
+    // TODO Push once all is good FIX
```

Comment:
Must be fixed before merge

Reply:
In other flows, all functions `handleInteractionToken` delegates to push the token at the end.

Reply:
Do we even need to collect the tokens on the Flow level, since we're already collecting them on the Interaction class? As far as I looked, this is not actually used anywhere (`this.tokens` on the flow level) besides pushing.

Reply:
Could be removed then.